mewarpx/mewarpx/assemblies.py

```diff
@@ -157,104 +183,95 @@ def record_scrapedparticles(self):
+                raw_particle_data['y'] = [struct['y'] for struct in structs]
+                raw_particle_data['z'] = [struct['z'] for struct in structs]
+            else:
+                raise NotImplementedError(
+                    f"Scraping not implemented for {mwxrun.geom_str}."
+                )
 
-                    idx_list = temp_idx_list
+            # sort the particles appropriately if this is an eb
+            if self.scraper_label == 'eb':
+                temp_idx_list = []
 
-            if not empty:
-                data[4] = sum(np.size(idx) for idx in idx_list)
-                w_arrays = mwxrun.sim_ext.get_particle_boundary_buffer(
-                    species.name, self.scraper_label, "w", mwxrun.lev
-                )
-                data[5] = -species.sq * sum(np.sum(w_arrays[i][idx_list[i]])
-                     for i in range(len(idx_list))
-                )
-                E_arrays = mwxrun.sim_ext.get_particle_boundary_buffer(
-                    species.name, self.scraper_label, "E_total", mwxrun.lev
-                )
-                data[6] = -sum(np.sum(E_arrays[i][idx_list[i]])
-                     for i in range(len(idx_list))
-                )
+                for i in range(len(idx_list)):
+                    is_inside = self.isinside(
+                        raw_particle_data['x'][i][idx_list[i]],
+                        raw_particle_data['y'][i][idx_list[i]],
+                        raw_particle_data['z'][i][idx_list[i]]
+                    )
+                    temp_idx_list.append(idx_list[i][np.where(is_inside)])
 
-            self.append_scrapedparticles(data)
+                    # set the scraped timestep to -1 for particles in this
+                    # EB so that they are not considered again
+                    comp_steps[i][idx_list[i][np.where(is_inside)]] = -1
 
-    def append_scrapedparticles(self, data):
-        """Append one or more lines of scraped particles data.
+                idx_list = temp_idx_list
 
-        Arguments:
-            data (np.ndarray): Array of shape (m) or (n, m) where m is the
-                number of fields and n is the number of rows of data to append.
-        """
-        self._scrapedparticles_data.append(data)
+            for ii, attrib in enumerate(self.scraped_particle_attribs_list):
+                # get the desired particle property
+                if attrib not in ['x', 'y', 'z']:
+                    raw_particle_data[attrib] = (
+                        mwxrun.sim_ext.get_particle_boundary_buffer(
+                            species.name, self.scraper_label, attrib, mwxrun.lev
+                        )
+                    )
 
-    def get_scrapedparticles(self, clear=False):
-        """Retrieve a copy of scrapedparticles data.
+            # loop over all tiles and append the particle data from that tile
+            # to scraped_particle_data
+            for ii, idx in enumerate(idx_list):
```

Comment:
`idxs` instead of `idx` perhaps? It threw me off - I think this is all particles on that tile that are collected, not just a single number.